lib/suggest.js

```diff
@@ -318,14 +336,19 @@ function generate(context, memory, words, edits) {
       while (++offset < characterLength) {
         inject = characters[offset]
 
-        // Add and replace.
-        check(before + inject + after)
-        check(before + inject + nextAfter)
-
         // Try upper-case if the original character was upper-cased.
         if (upper && inject !== inject.toUpperCase()) {
+          if (currentCase !== 's') {
+            check(before + inject + after)
+            check(before + inject + nextAfter)
+          }
```

Comment:
This avoids double weighting of lowercase words when both lowercase and sentence case are tested for the same word.

Without this line, the "dont" test would artificially favor results such as "font" over results such as "dent" because "font" would be counted twice.